crates/networking/discv5/src/subnet.rs

```diff
@@ -0,0 +1,149 @@
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Subnet {
+    Attestation(u8),
+    SyncCommittee(u8),
+}
+
+#[derive(Clone, Debug, Default, Encode, Decode)]
+pub struct Subnets {
+    attestation_bits: Option<BitVector<U64>>,
+}
+
+impl Subnets {
+    pub fn new() -> Self {
+        Self {
+            attestation_bits: None,
+        }
+    }
+
+    pub fn enable_subnet(&mut self, subnet: Subnet) -> anyhow::Result<()> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                let bits = self.attestation_bits.get_or_insert(BitVector::new());
+                bits.set(id as usize, true)
+                    .map_err(|_| anyhow!("Subnet ID out of bounds"))?;
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn disable_subnet(&mut self, subnet: Subnet) -> anyhow::Result<()> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                if let Some(bits) = &mut self.attestation_bits {
+                    bits.set(id as usize, false)
+                        .map_err(|_| anyhow!("Subnet ID out of bounds"))?;
+                }
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn is_active(&self, subnet: Subnet) -> bool {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => self
+                .attestation_bits
+                .as_ref()
+                .is_some_and(|bits| bits.get(id as usize).unwrap_or(false)),
+            Subnet::Attestation(_) => false,
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+}
+
+impl Encodable for Subnets {
+    fn encode(&self, out: &mut dyn BufMut) {
+        let ssz_bytes = self.as_ssz_bytes();
+        let bytes = Bytes::from(ssz_bytes);
+        bytes.encode(out);
+    }
+}
+
+impl Decodable for Subnets {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        let bytes = Bytes::decode(buf)?;
+        let subnets = Subnets::from_ssz_bytes(&bytes)
+            .map_err(|_| alloy_rlp::Error::Custom("Failed to decode SSZ subnets"))?;
+        Ok(subnets)
+    }
+}
+
+pub fn subnet_predicate(subnets: Vec<Subnet>) -> impl Fn(&Enr) -> bool + Send + Sync {
+    move |enr: &Enr| {
+        let value = if let Some(result) = enr.get_decodable::<Bytes>(ATTESTATION_BITFIELD_ENR_KEY) {
+            match result {
+                Ok(value) => value,
+                Err(_) => return false,
+            }
+        } else {
+            return false;
+        };
+        let subnets_state = match Subnets::from_ssz_bytes(&value) {
+            Ok(subnets) => subnets,
+            Err(_e) => return false,
+        };
```

Comment:
```suggestion
        let Some(Ok(subnets_state)) = enr.get_decodable::<Subnets>(ATTESTATION_BITFIELD_ENR_KEY) else {
            return false
        } 
```

This can be simplified to the above, since we now implement the Decodable trait

Reply:
oh, that's great , i didn't realise that we can get subnet object directly from get_decodable, i will keep in mind for future references, thanks